src/components/Offers/EditOfferForm.js

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback, useContext } from "react";
+import { parseRequestErrors } from "./New/CreateOfferUtils";
+import OfferForm from "./New/form-components/OfferForm";
+import { editOffer } from "../../services/offerService";
+import { Redirect, useParams } from "react-router-dom";
+import useOffer from "../../hooks/useOffer";
+import useOfferForm from "../../hooks/useOfferForm";
+
+export const EditOfferControllerContext = React.createContext();
+
+export const EditOfferController = () => {
+    const { id } = useParams();
+    const { offer, error: errorOffer, loading: loadingOffer } = useOffer(id);
+    const redirectProps = { to: { pathname: "/not-found" } };
+
+    const params = useOfferForm(offer);
```

Comment:
Because of the way these hooks work, i can't set the default values in useForm default values. How can I do this?

Reply:
why not? Looking at the code I don't see why it would fail

Reply:
When the first useOfferForm call is executed, the useOffer request is still loading, hence offer is undefined and defaultValues of the useForm hook inside offerForm will be set to the default

Reply:
That's easy. Move the `useOfferForm` call to another child component, that is only rendered *after* we have an offer/we are done loading. That way, we only render the form once we have the offer details, which makes sense. If we don't (in case of an error), we should show an error message, and it doesn't make sense to have a form anyway.